jibri/rootfs/defaults/jibri.conf

```
@@ -0,0 +1,100 @@
+{{ $ENABLE_JIBRI_HTTP_API_PORT := .Env.ENABLE_JIBRI_HTTP_API_PORT | default "1" | toBool }}
+jibri {
+    // A unique identifier for this Jibri
+    // TODO: eventually this will be required with no default
+    id = "jibri-instance-{{ .Env.JIBRI_INSTANCE_ID }}"
+    
+    api {
+        {{ if $ENABLE_JIBRI_HTTP_API_PORT -}}
+        http {
+            {{  if .Env.JIBRI_HTTP_API_EXTERNAL_PORT -}}
+            external-api-port = {{ .Env.JIBRI_HTTP_API_EXTERNAL_PORT }}
+            {{ end -}}
+            {{ if .Env.JIBRI_HTTP_API_INTERNAL_PORT -}}
+            internal-api-port = {{ .Env.JIBRI_HTTP_API_INTERNAL_PORT }}
+            {{ end -}}
```

this is more clean approach. If any of the value is not configured then this will fallback to reference.conf.
```suggestion
      {{ if and .Env.JIBRI_HTTP_API_EXTERNAL_PORT .Env.JIBRI_HTTP_API_INTERNAL_PORT }}
      http {
        external-api-port = {{ .Env.JIBRI_HTTP_API_EXTERNAL_PORT }}       
        internal-api-port = {{ .Env.JIBRI_HTTP_API_INTERNAL_PORT }}
      }
      {{ end }}
```

---

What if the user want's to configure only `external-api-port` and not the `internal-api-port`.  If I am not mistaken, `and` logic doesn't provide this feature here. What do you think?

---

yes it doen't. 

Since there is no enable/disable here hence even if you set `ENABLE_JIBRI_HTTP_API_PORT =0` then it'll be served from reference.conf hence there is no purpose of using this env variable. 

we have to set the default values of both of the env variables i.e `JIBRI_HTTP_API_EXTERNAL_PORT=3333` and `JIBRI_HTTP_API_INTERNAL_PORT=2222` in env.example/.env files. Lets say you changed `JIBRI_HTTP_API_EXTERNAL_PORT =4444` but didn't change `JIBRI_HTTP_API_EXTERNAL_PORT ` then  `JIBRI_HTTP_API_INTERNAL_PORT ` will be served from `env.example/.env` file.

---

@saghul Can you please suggest which approach is better here?

---

I'm tempted to agree with what @saygun's point about having the flexibility to override only a single one of the ports, but @prayagsingh I'm not sure I followed what you were saying there.

---

Not sure I followed either. I think they should be settable independently and you can use `| default "0" | toBool` to check if it got disabled by setting it to empty or 0, for example.

---

@saghul @bbaldino  I'm trying to avoid an extra env variable. I have updated my suggestion. Please check.

---

Looks good!